internal/iterator_wrapper.hpp

```diff
@@ -32,15 +32,15 @@ namespace iter {
 
     template <typename Container>
     using IteratorWrapper = typename IteratorWrapperImplType<Container,
-        std::is_same<impl::iterator_type<Container>,
-            impl::iterator_end_type<Container>>{}>::type;
+        std::is_same_v<impl::iterator_type<Container>,
```

Comment:
is this actually required for msvc?

Reply:
Yes, it is.  I should report the bug in msvc.  It gives an "error C2512: 'std::is_same<Ts,Ts>': no appropriate default constructor available".  There is a problem with the constexpr contructor when used in a variadic template.  But anyway, since C++17, I would say that _v template variables are the preferred way, thus this modification follows C++17.